### PR TITLE
Promote gateway injection to beta

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

File: features/gateway_injection.md [hunk 1]
```diff
@@ -92,19 +92,19 @@
 
 **Tests**
 
-- [ ] Integration tests cover feature edge cases
+- [x] Integration tests cover feature edge cases
 - [x] End-to-end tests cover samples/tutorials. Documentation has page tests.
 - [x] Fixed issues have tests to prevent regressions. There are no known issues with this feature.
-- [ ] Stability/stress test suite includes coverage for the feature.
+- [x] Stability/stress test suite includes coverage for the feature.
```
Comment (howardjohn) [inline features/gateway_injection.md:98]: Not accurate yet; will send PR soon

File: features/gateway_injection.md [hunk 2]
```diff
@@ -92,19 +92,19 @@
...
 
-- [ ] Feature coverage and test plans written and approved
-- [ ] Tests exist with the feature enabled that can be integrated with our automated performance testing.
+- [x] Feature coverage and test plans written and approved
+- [x] Tests exist with the feature enabled that can be integrated with our automated performance testing.
 
 **API**
 
-- [ ] TOC has reviewed the API and determined it to be complete.
+- [x] TOC has reviewed the API and determined it to be complete.
```
Comment (howardjohn) [inline features/gateway_injection.md:107]: Pending approval on this PR

File: features/gateway_injection.md [hunk 3]
```diff
@@ -116,9 +116,9 @@
 
 **Approvals**
 
-- [ ] The appropriate work group(s) have reviewed and approved promotion of the feature.
-- [ ] The supportability review panel has reviewed promotion of the feature.
-- [ ] The TOC has reviewed and approved promotion of the feature as part of the
+- [x] The appropriate work group(s) have reviewed and approved promotion of the feature.
```
Comment (howardjohn) [inline features/gateway_injection.md:119]: Pending approval on this PR

File: features/gateway_injection.md [hunk 4]
```diff
@@ -116,9 +116,9 @@
...
+- [ ] The supportability review panel has reviewed promotion of the feature. N/A. Feature is too minor to involve panel
+- [x] The TOC has reviewed and approved promotion of the feature as part of the
```
Comment (howardjohn) [inline features/gateway_injection.md:121]: Pending approval on this PR

File: features/gateway_injection.md (hunk 1 above)
Comment (howardjohn) [reply features/gateway_injection.md:98]: Done: https://github.com/istio/tools/pull/1673

**Comment by craigbox** [comment]
Just to confirm: is this for 1.12.0, 1.11.0 or a 1.11 point release?

**Comment by howardjohn** [comment]
> Just to confirm: is this for 1.12.0, 1.11.0 or a 1.11 point release?

Whatever will get approved I suppose :slightly_smiling_face: . There has not been any code changes to this feature since it was introduced as experimental, so there is no specific releases it needs to be/not be in

**Comment by ericvn** [comment]
My comment on this would be since it's in the master/main branch, the feature will show up as `beta` in the 1.12 version of preliminary.istio.io after 1.11 is released. If we wanted it to show in the 1.11 version of istio.io (current preliminary.istio.io), we would need to cherry pick the change to the release-1.11 branch.

**Comment by craigbox** [comment]
If there's no change to the code required, and we meet the bar today - the only change is a line item in the docs - I vote we get it in for 1.11!

**Comment by brian-avery** [comment]
+1. If we want this promoted in 1.11, we should cherrypick. And I don't think we want to promote features in point releases, even if there's no code change. Which means that we should do it for 1.11.0 if we're going to.

**Comment by brian-avery** [comment]
In the face of zero code changes, this really amounts to a docs change. @istio/technical-oversight-committee thoughts on promoting this before we release 1.11?

**Comment by linsun** [comment]
@brian-avery main concern is feedback from user is lacking and still very new feature.  I think 1.12 beta is reasonable.

**Comment by howardjohn** [comment]
@brian-avery @linsun ping

**Comment by smawson** [comment]
Even though you didn't ping me I'm good with shipping it. But we missed getting it into 1.11.0 so...are we ok retroactively marking it beta? If no code changes I think I'm OK with that.

**Comment by howardjohn** [comment]
1.12 is coming up, lets just upgrade it for that release now

File: features.yaml [hunk 5]
```diff
@@ -38,8 +38,8 @@ features:
     id: :"traffic.gateway_injection"
     level:
       checklist: features/gateway_injection.md
-      maturity: Alpha
-      nextExpectedPromotion: "1.11"
+      maturity: Beta
+      nextExpectedPromotion: "1.13"
```
Comment (ericvn) [inline features.yaml:42]: Since this is going to be Beta in 1.12, based on an approval now, do we want to change the next expected promotion to 1.14? Otherwise, LGTM.
